docs/advanced/github_actions.md

```diff
@@ -235,6 +235,29 @@ If you want to use PowerShell, you need to specify `-Command` as well.
     There are some custom shells provided by GitHub that have slightly different behavior, see [`jobs.<job_id>.steps[*].shell`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) in the documentation.
     See the [official documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#custom-shell) and [ADR 0277](https://github.com/actions/runner/blob/main/docs/adrs/0277-run-action-shell-options.md) for more information about how the `shell:` input works in GitHub Actions.
 
+#### One-off shell wrapper using `pixi exec`
+
+With `pixi exec`, you can also run a one-off command inside a temporary pixi environment.
+
+```yaml
+- run: | # (1)!
+    zstd --version
+  shell: pixi exec --spec zstd -- bash -e {0}
+```
+
+1. everything here will be run inside of the temporary pixi environment
+
+```yaml
+- run: | # (1)!
+    import ruamel.yaml
+    # ...
+  shell: pixi exec --spec python=3.11.* --spec ruamel.yaml -- python {0}
```

Comment:
small note: `'python=3.11.*'` doesn't work here as this isn't evaluated in a shell and it would be passed with quotes to pixi which results in the following error:

```
error: invalid value ''python=3.11.*'' for '--spec <SPECS>': ''python' is not a valid package name. Package names can only contain 0-9, a-z, A-Z, -, _, or .

For more information, try '--help'.
```

Nothing we need to fix on pixi's side IMO